api/python/provisioner/commands/validator/validator.py

```diff
@@ -784,3 +787,128 @@ def __attrs_post_init__(self):
+            The first operand
+        b: str
+            The second operand
+        operator: str
+            String representation of the operator
+
+        Returns
+        -------
+        bool:
+            return True if comparison is true and False otherwise
+
+        """
+        operator = ComparisonOperators(operator)
+
+        if operator == ComparisonOperators.GREATER:
+            return version.parse(a) > version.parse(b)
+        elif operator == ComparisonOperators.LOWER:
+            return version.parse(a) < version.parse(b)
+        elif operator == ComparisonOperators.EQUAL:
+            return version.parse(a) == version.parse(b)
+        elif operator == ComparisonOperators.NOT_EQUAL:
+            return version.parse(a) != version.parse(b)
+        elif operator == ComparisonOperators.GREATER_OR_EQUAL:
+            return version.parse(a) >= version.parse(b)
+        elif operator == ComparisonOperators.LOWER_OR_EQUAL:
+            return version.parse(a) <= version.parse(b)
+        else:
+            raise ValidationError(
+                f"Unexpected comparison operator: '{operator}'")
+
+    def validate(self, iso_info):
+        """
+
+        Parameters
+        ----------
+        iso_info: CortxISOInfo
+            CortxISOInfo instance with all necessary information about
+            SW upgrade ISO
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        ValidationError
+            If validation is failed.
+        """
+        packages = list(iso_info.packages.keys())
+
+        # NOTE: the first line of `yum -q list installed` command is
+        #  'Installed Packages' skip it via `tail -n +2`
+        cmd = (f"yum -q list installed {packages} | tail -n +2 | "
+               "awk '{print $1\" \"$2}'")
+
+        try:
+            res = cmd_run(cmd, targets=local_minion_id())
+        except Exception as e:
+            logger.debug(f'Package compatibility check is failed: "{e}"')
+            raise ValidationError(
+                f'Package compatibility check is failed: "{e}"') from e
+
+        res = res[local_minion_id()].strip()
+
+        if res:
+            logger.debug(f"List of installed CORTX packages: {res}")
+        else:
+            logger.error(f"There are no installed CORTX packages")
+
+        res = res.split('\n')
+
+        packages = dict()
+        for pkg in res:
+            # Aggregate version information of installed CORTX packages
+            pkg_name, version = pkg.split(" ")
+            packages[pkg_name] = version
+
+        error_msg = list()
+        for pkg in iso_info.packages:
+            if (SWUpgradeInfoFields.VERSION_COMPATIBILITY.value
+                    in iso_info.packages[pkg]):
+                compatibility = iso_info.packages[pkg][
+                    SWUpgradeInfoFields.VERSION_COMPATIBILITY.value]
+
+                installed_ver = packages.get(pkg, None)
+                if installed_ver is None:
```

Comment:
if a package is not installed it should be ok to install any new version, so it's not an error

Reply:
Done.